lib/node.py

```diff
@@ -0,0 +1,382 @@
+import os
+import time
+import sys
+import signal
+import subprocess
+import logging
+import time
+import socket
+
+log = logging.getLogger(__name__)
+
+
+def shellcmd(**kwargs):
+    username = kwargs.get('user', 'jenkins')
+    command = kwargs.get('cmd', None)
+    ipaddr = kwargs.get('host', None)
+    res, out, err = ssh(cmd=command, host=ipaddr, user=username)
+    if res != 0:
+        msg = "Error while executing the command '%s'. Error message: '%s'" % (command, err)
+        raise(Exception, msg)
+    return None
+
+
+def ssh(**kwargs):
+    username = kwargs.get('user', None)
+    hostname = kwargs.get('host', None)
+    command  = kwargs.get('cmd', None)
+    timeout  = kwargs.get('timeout', None)
+    workdir  = kwargs.get('cwd', None)
+    environ  = kwargs.get('env', None)
+    cmd = "ssh -q %s@%s %s" % (username, hostname, command)
+    return launch(cmd=cmd, timeout=timeout, env=environ, cwd=workdir)
+
+
+def launch(**kwargs):
+    orig_wd = None
+    execution_string = kwargs.get('cmd', None)
+    env = kwargs.get('env', None)
+    timeout = kwargs.get('timeout', None)
+    workdir = kwargs.get('cwd', None)
+    quiet   = kwargs.get('quiet', False)
+    if env is None:
+        env = os.environ
+    if timeout is None:
+        timeout = 60
+
+    if execution_string is None:
```

Comment:
`if not execution_string` would be better. That would also trigger when it's an empty string